packages/webpack/src/utils/module-federation/share.ts

```diff
@@ -11,20 +11,30 @@ import {
   collectPackageSecondaryEntryPoints,
   collectWorkspaceLibrarySecondaryEntryPoints,
 } from './secondary-entry-points';
-import { type ProjectGraph, workspaceRoot, logger } from '@nx/devkit';
+import {
+  type ProjectGraph,
+  workspaceRoot,
+  logger,
+  readJsonFile,
+  ProjectGraphProjectNode,
+  joinPathFragments,
+} from '@nx/devkit';
+import { existsSync } from 'fs';
+import type { PackageJson } from 'nx/src/utils/package-json';
 
 /**
  * Build an object of functions to be used with the ModuleFederationPlugin to
  * share Nx Workspace Libraries between Hosts and Remotes.
  *
- * @param libraries - The Nx Workspace Libraries to share
+ * @param workspaceLibs - The Nx Workspace Libraries to share
  * @param tsConfigPath - The path to TS Config File that contains the Path Mappings for the Libraries
  */
 export function shareWorkspaceLibraries(
-  libraries: WorkspaceLibrary[],
+  workspaceLibs: WorkspaceLibrary[],
+  project: ProjectGraphProjectNode['data'],
```

Comment:
Better to change this to `projectRoot: string` I think. More flexible, and not tied to the project node specifically. Also, can it be made required? Not sure if there is a case where it doesn't need to be passed in.

Reply:
Will update this thanks